src/pages/Products/Products.js

```diff
@@ -33,26 +42,47 @@ export function Products() {
   }
 
   useEffect(() => {
-    searchParams.set('page', currentPage);
+    if (!searchValue) {
+      searchParams.delete("search");
+      setSearchParams(searchParams);
+    }
+    loadProducts();
+  }, [searchValue]);
+
+  useEffect(() => {
+    searchParams.set("page", currentPage);
     setSearchParams(searchParams);
-    loadPosts();
+    setProducts(showVisibleProducts(totalProducts, productsPerPage, currentPage));
   }, [currentPage]);
 
   return (
     <div className="Products">
       <div className="Products__container container">
-        <h2 className="Products__title">Dogs</h2>
+        <div className="Products__top-row">
+          <h2 className="Products__title">Dogs</h2>
+          <SearchField instantChangeProducts={setSearchValue} />
+        </div>
+        {searchValue && !products.length && !loading && !isError && (
+          <ErrorMessage message="There are no dogs with this breed" />
+        )}
         {isError && <ErrorMessage />}
         {loading && <Preloader />}
         <div className="Products__catalog">
-          {products &&
+          {!!products.length &&
```

Comment:
it's better to check on products && products.length > 0 instead (or simply products?.length > 0)